### PR TITLE
Added "consortium" search index. and modified all other annotation-based indicies to document only 1 annotation value (PLFM-4754)

### PR DESCRIPTION
**Pull request by zimingd**

### CODE REVIEW

**Review by meredithslota — COMMENTED** [review]
1. In /src/main/resources/schema/org/sagebionetworks/repo/model/search/DocumentFields.json line #81: the reference URL for 'tissue' (http://synapse.sagebase.org/ontology#6501) is incorrect and doesn't exist anymore; can we omit?

2. Same file, line #108: the description is "The name of the consortium" from https://github.com/Sage-Bionetworks/synapseAnnotations/blob/master/synapseAnnotations/data/sageCommunity.json line #66 but we may want to modify it later on.

File: lib/lib-search/src/test/java/org/sagebionetworks/search/SearchSchemaLoaderTest.java [hunk 1]
```diff
@@ -17,8 +17,8 @@
 	public void testLoadSearchDomainSchema() throws IOException{
 		List<IndexField> list = SearchSchemaLoader.loadSearchDomainSchema();
 		assertNotNull(list);
-		// We currently have 25 index fields
-		assertEquals(25, list.size());
+		// We currently have 19 index fields
```
Comment (zimingd) [inline lib/lib-search/src/test/java/org/sagebionetworks/search/SearchSchemaLoaderTest.java]: 18

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java [hunk 2]
```diff
@@ -294,71 +238,78 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
 	}
 
 
-	static void addAnnotationsToSearchDocument(DocumentFields fields,
-			Annotations annots) {
+	void addAnnotationsToSearchDocument(DocumentFields fields, Annotations annots) {
```
Comment (zimingd) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java]: use Annotations.getSingleValue(). rename to getFirstValue(). and reverse the mapping to index field->annotation names.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java [hunk 3]
```diff
@@ -294,71 +238,78 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
...
-					try {
-						fields.getNum_samples().add(
-								Long.valueOf((stringValue).trim()));
+	/**
+	 * Checks the key and value of
+	 * @param fields DocumentField to which the annotation is added
+	 * @param key Annotation key
+	 * @param value Annotation value
+	 */
+	void addAnnotationToDocumentField(DocumentFields fields, String key, Object value) {
```
Comment (zimingd) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java]: combine with above method. 

Sequentially set each document field if it is contained in the Annotations. get rid of case switch.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java [hunk 4]
```diff
@@ -292,76 +231,67 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
...
+
+		//set the values for the document fields
+		fields.setDisease(getSearchIndexFieldValue(firstAnnotationValues, FIELD_DISEASE));
+		fields.setConsortium(getSearchIndexFieldValue(firstAnnotationValues, FIELD_CONSORTIUM));
+		fields.setTissue(getSearchIndexFieldValue(firstAnnotationValues, FIELD_TISSUE));
+		fields.setPlatform(getSearchIndexFieldValue(firstAnnotationValues, FIELD_PLATFORM));
+		try {
+			fields.setNum_samples(NumberUtils.createLong(getSearchIndexFieldValue(firstAnnotationValues, FIELD_NUM_SAMPLES)));
+		}catch (NumberFormatException e){
+			// swallow this exception, this is just a best-effort
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java:246]: if the users did not provide a numeric value for "number of samples" then that value will not be added to the search index but is not an error.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java [hunk 5]
```diff
@@ -292,76 +231,67 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
...
-				if (null == objs[i])
-					continue;
+	/**
+	 * Returns a Map from the keys of the NamedAnnotations to the first value (as a String) for that key.
+	 * @param annotations
+	 * @return
+	 */
+	Map<String, String> getFirsAnnotationValues(NamedAnnotations annotations){
+		Map<String, String> firstAnnotationValues = new HashMap<>();
+		addFirstAnnotationValuesToMap(annotations.getPrimaryAnnotations(), firstAnnotationValues);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java:259]: additional only.

File: services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java [hunk 6]
```diff
@@ -292,76 +231,67 @@ public Document formulateSearchDocument(Node node, NamedAnnotations annos,
...
-								+ " added to searchable annotations map but not added to addAnnotationToSearchDocument");
+	/**
+	 * Given the index field name, retrieve
+	 * @param firsAnnotationValues
+	 * @param indexFieldKey
+	 * @return
+	 */
+	String getSearchIndexFieldValue(Map<String, String> firsAnnotationValues, String indexFieldKey){
+		for(String possibleAnnotationName: SEARCHABLE_NODE_ANNOTATIONS.get(indexFieldKey)){
+			String value = firsAnnotationValues.get(possibleAnnotationName);
```
Comment (john-hill) [inline services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/SearchDocumentDriverImpl.java:286]: toLower() and test.

**Pull request merged**